### PR TITLE
Use the user's default currency when choosing which one to send

### PR DESCRIPTION
**Pull request by zanderz**

### CODE REVIEW

File: shared/reducers/wallets.js [hunk 1]
```diff
@@ -41,7 +41,9 @@ export default function(state: Types.State = initialState, action: WalletsGen.Ac
     case WalletsGen.displayCurrenciesReceived:
       return state.set('currencies', I.List(action.payload.currencies))
     case WalletsGen.displayCurrencyReceived:
-      return state.setIn(['currencyMap', action.payload.accountID], action.payload.currency)
+      return state
+        .setIn(['currencyMap', action.payload.accountID], action.payload.currency)
+        .set('buildingPayment', state.get('buildingPayment').merge({currency: action.payload.currency.code}))
```
Comment (chrisnojima) [inline shared/reducers/wallets.js]: use `.update`

Comment (zanderz) [reply shared/reducers/wallets.js]: like this?

**Comment by zanderz** [comment]
ready again after some confusing merging

File: shared/reducers/wallets.js [hunk 2]
```diff
@@ -50,7 +50,9 @@ export default function(state: Types.State = initialState, action: WalletsGen.Ac
     case WalletsGen.displayCurrenciesReceived:
       return state.set('currencies', I.List(action.payload.currencies))
     case WalletsGen.displayCurrencyReceived:
-      return state.setIn(['currencyMap', action.payload.accountID], action.payload.currency)
+      return state
+        .update('currencyMap', c => c.set(action.payload.accountID, action.payload.currency))
+        .set('building', state.get('building').merge({currency: action.payload.currency.code}))
```
Comment (chrisnojima) [inline shared/reducers/wallets.js]: this can use update also

Comment (zanderz) [reply shared/reducers/wallets.js]: ok

**Pull request merged**